pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -141,6 +147,16 @@ func (w *Webhook) WebhookFunc() admission.WebhookFunc {
 	}
 }
 
+// isReadOnlyRootFilesystem returns whether the agent sidecar should have readOnlyRootFilesystem security setup
+// This will always activate unless a profile is provided with a securityContext that has readOnlyRootFilesystem set to false
+func (w *Webhook) isReadOnlyRootFilesystem() bool {
+	if w.profileOverrides == nil || len(*w.profileOverrides) == 0 {
+		return true
+	}
+	securityContext := (*w.profileOverrides)[0].SecurityContext
+	return securityContext == nil || securityContext.ReadOnlyRootFilesystem == nil || *securityContext.ReadOnlyRootFilesystem
```

Comment:
I'm not sure I'm following here. 

If `securityContext == nil`, we would we return true? If we have no security context then this is not a ReadOnlyFilteSystem by default

Reply:
If the user hasn't defined any of these values, then we want to default to true. I can more explicitly state that we want the default value to be true in the function's comment description.